src/main/java/com/commercetools/sync/commons/helpers/BaseReferenceResolver.java

```diff
@@ -68,40 +65,12 @@ protected BaseReferenceResolver(@Nonnull final S options) {
     protected static String getKeyFromResourceIdentifier(@Nonnull final ResourceIdentifier resourceIdentifier,
                                                          final boolean allowUuidKeys)
         throws ReferenceResolutionException {
+
         final String key = resourceIdentifier.getId();
```

Comment:
We get ID from the resourceIdentifier and name it as a key. However there is a key field on the resourceIdentifier too which makes everything confusing. Could you add a comment explaining why it's used this way?
Sorry for out of scope comment 🐳

Reply:
a099c2d